### PR TITLE
Module2 project exercise

### PR DESCRIPTION
**Pull request by jainsushma**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

**Comment by jainsushma** [comment]
Draft discussion is finishedd

**Comment by jainsushma** [comment]
Can you please review my PR?

File: todo_app/trello_board_actions.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        else:
+            new_list_id = Trello_Board_Actions.getListIdByName(self, "to-do")
+       
+        self.payload = self.auth.copy()
+        self.payload['idList'] = new_list_id
+        response = requests.put(url=self.url + '/cards/' + id, data=self.payload)
+        if response.status_code == 200:
+            result_flag = True
+
+        return result_flag
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]: Defining a result_flag variable is overcomplicated. Why not just `return response.status_code == 200`?

File: todo_app/trello_board_actions.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            result_flag = True
+        return result_flag
+
+    def getListIdByName(self,list_name):
+        "Get the list id"
+        lists_on_board = Trello_Board_Actions.getBoardLists(self)
+        for i in range(len(lists_on_board)):
+            if lists_on_board[i]['name'] == list_name:
+                list_id = lists_on_board[i]['id']
+        return list_id
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]:
```suggestion
        for list_on_board in lists_on_board:
            if list_on_board['name'] == list_name:
                return list_on_board['id']
        return None
```

File: todo_app/trello_board_actions.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                }
+
+    def getBoardLists(self):
+        "Get the lists for a given board id"
+        board_lists = requests.get(url=self.url + '/boards/' + os.getenv('BOARD_ID') +'/lists', params=self.auth)
+        return board_lists.json()
+
+    def getCards(self):
+        "Get the card on a list and its status"
+        board_lists = Trello_Board_Actions.getBoardLists(self)
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]:
```suggestion
        board_lists = self.getBoardLists()
```

File: todo_app/trello_board_actions.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    def __init__(self):
+        self.auth = {'key':os.getenv("SECRET_KEY"), 'token':os.getenv("TOKEN"), 'cards':"open"}
+        self.url= "https://api.trello.com/1"
+        self.headers = {
+            'type': "type",
+            'content-type': "application/json"
+                }
+
+    def getBoardLists(self):
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]: Naming convention is snake_case for variable and function names; PascalCase for class names

File: todo_app/app.py [hunk 5]
```diff
@@ -1,24 +1,32 @@
...
+    Trello_Board_Actions().addNewCard(title)
+    return redirect(os.getenv("TODO_HOSTNAME"))
+
+@app.route('/move_item/<id>', methods=['POST'])
+def move_item(id):
+    Trello_Board_Actions().updateCardStatus(id)
+    return redirect(os.getenv("TODO_HOSTNAME"))
+
+@app.route('/delete_item/<id>', methods=['POST'])
+def delete_item(id):
```
Comment (hugh-emerson) [inline todo_app/app.py:27]: Nothing points to this route

File: todo_app/templates/index.html [hunk 6]
```diff
@@ -7,20 +7,61 @@ <h1 class="display-4">To-Do App</h1>
...
     <div class="col-md-12">
       <h2>Items</h2>
-      <ul class="list-group mb-4">
+      <ul class="list-group mb-4" style="list-style-type:none;">
         <!-- List the todo items here -->
         {% for item in items %}
-          <li>{{ item.title }} - {{item.status}}</li>
+          <li>
+            <div class="col-sm-10">
+              <div style="text-align:left">
```
Comment (hugh-emerson) [inline todo_app/templates/index.html:30]: Putting a div inside a div inside the li shouldn't be necessary.

File: todo_app/trello_board_actions.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+class Trello_Board_Actions:
+    " Trello util to get lists on a board, get cards and their statuses, update/delete a card"    
+
+    def __init__(self):
+        self.auth = {'key':os.getenv("SECRET_KEY"), 'token':os.getenv("TOKEN"), 'cards':"open"}
+        self.url= "https://api.trello.com/1"
+        self.headers = {
+            'type': "type",
+            'content-type': "application/json"
+                }
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]: This is unused.

File: todo_app/trello_board_actions.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+                            )
+                items.append(card)
+        return items
+    
+    def get_items(self):
+        "Simple attempt to get all cards from Trello."
+        response = requests.get(f"https://api.trello.com/1/boards/{os.getenv('BOARD_ID')}/cards", params=self.auth)
+        todos = response.json()
+        # for card in todos:
+        #     print(card['name'], card['status'])
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]: Delete commented out code

File: todo_app/trello_board_actions.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                list_id = lists_on_board[i]['id']
+        return list_id 
+
+    def isCardOnList(self, list_name, card_id):
+        "Check if the card is on given list name"
+        cards_on_the_list = requests.get(url=self.url + '/lists/' + Trello_Board_Actions.getListIdByName(self, list_name) + '/cards', params=self.auth)
+        cards_on_the_list = cards_on_the_list.json()
+        for i in range(len(cards_on_the_list)):
+            if cards_on_the_list[i]['id'] == card_id:
+                return True
```
Comment (hugh-emerson) [inline todo_app/trello_board_actions.py]:
```suggestion
        for card in cards_on_the_list:
            if card['id'] == card_id:
                return True
```

File: todo_app/data/session_items.py [hunk 10]
```diff
@@ -1,11 +1,11 @@
 from flask import session
```
Comment (hugh-emerson) [inline todo_app/data/session_items.py]: You can now delete this file

File: README.md [hunk 11]
```diff
@@ -30,7 +30,13 @@ You'll also need to clone a new `.env` file from the `.env.template` to store lo
 $ cp .env.template .env  # (first time only)
 ```
 
-The `.env` file is used by flask to set environment variables when running `flask run`. This enables things like development mode (which also enables features like hot reloading when you make a file change). There's also a [SECRET_KEY](https://flask.palletsprojects.com/en/1.1.x/config/#SECRET_KEY) variable which is used to encrypt the flask session cookie.
+The `.env` file is used by flask to set environment variables when running `flask run`. This enables things like development mode (which also enables features like hot reloading when you make a file change). 
+
+In order to run this application you will require the appropriate trello information in the .env file. They are as follows
```
Comment (hugh-emerson) [inline README.md:35]: You also need a board with three specific list names.

**Review by hugh-emerson — APPROVED** [review]
Looks good. I've left some suggestions but merge when you're ready. 👍

File: todo_app/data/session_items.py (hunk 10 above)
Comment (jainsushma) [reply todo_app/data/session_items.py]: Done

File: todo_app/trello_board_actions.py (hunk 2 above)
Comment (jainsushma) [reply todo_app/trello_board_actions.py]: done

File: todo_app/app.py (hunk 5 above)
Comment (jainsushma) [reply todo_app/app.py:27]: @hugh-emerson , sorry couldn't understand what is needed here

File: README.md (hunk 11 above)
Comment (jainsushma) [reply README.md:35]: Added

**Pull request merged**